psh/runfile/runfile.c

```diff
@@ -80,7 +82,19 @@ int psh_runfile(int argc, char **argv)
 	/* Take back terminal control */
 	tcsetpgrp(STDIN_FILENO, getpgid(getpid()));
 
-	return ((pid > 0) && (ret >= 0)) ? WEXITSTATUS(status) : EXIT_FAILURE;
+	if (!((pid > 0) && (ret >= 0))) {
+		retval = EXIT_FAILURE;
+	}
+	else if (WIFSIGNALED(status) != 0) {
+		/* Checking for the signal termination */
+		retval = SIGNAL_SHIFT + WTERMSIG(status);
```

Comment:
Ok, from what I gather signal that causes death should be shifted left on the status mask, but also WTERMSIG() should handle this fact, so it seems to me that there should not be this `SIGNAL_SHIFT +` part. I also don't like idea of this being defined here, such information should come from stdlib.

Reply:
IMHO this is posix-shell specific, at least busybox ash (https://github.com/mirror/busybox/blob/master/shell/ash.c#L4638) and bash (https://github.com/bminor/bash/blob/15409324f1974d41c183904ad575da7188058c1c/jobs.c#L3068) does it in userspace.

I don't see 128 added in libphoenix (https://github.com/phoenix-rtos/libphoenix/blob/25904ba0bbbec528fc0e9a0fcb8370044f5df4c1/signal/signal.c#L62) so I assume we're doing it the same way.

Edit:
I've verified that it's specified in POSIX, in shell-related chapters. Other special return codes are specified - which we could easily implement in `psh` to make it more posixly: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_08_02